gems/aws-sdk-core/spec/aws/credential_resolution_chain_spec.rb

```diff
@@ -818,5 +1188,9 @@ def stub_token_file(token)
       allow(File).to receive(:read).and_call_original
       allow(File).to receive(:read).with('my-token.jwt').and_return(token)
     end
+
+    def metrics_from_user_agent_header(header)
+      header[(header.index('m/') + 2)..-1]
```

Comment:
Try something like this maybe?

```
metrics = header.match(/ m\/([A-Za-z0-9+-,]+)/)[1].split(',')
```

The regex I believe covers all of the available metrics (check the business metric encoding design). Plug it into regexr.com and verify. It should match a space, then m/, then any of those combinations of metrics, and then a comma to separate, as a capture group. Then we retrieve that group and split, and it should be a native array.